src/core/ext/xds/xds_api.cc

```diff
@@ -866,7 +866,7 @@ absl::StatusOr<XdsApi::ResourceName> ParseResourceNameInternal(
   // Old-style names use the empty string for authority.
   // ID is prefixed with "old:" to indicate that it's an old-style name.
   if (!absl::StartsWith(name, "xdstp:")) {
-    return XdsApi::ResourceName{"", absl::StrCat("old:", name)};
+    return XdsApi::ResourceName{"old:", absl::StrCat("old:", name)};
```

Comment:
Since we're now going to have the `old:` or `xdstp:` prefix in the authority, I don't think we need it in the id field anymore.  So this can be:

```
return XdsApi::ResourceName{"old:", name};
```

Reply:
done